bug while parsing github json

```
Exception in thread "main" java.io.IOException: Parse exception converting JSON to object
    at org.eclipse.egit.github.core.client.GitHubClient.parseJson(GitHubClient.java:424)
    at org.eclipse.egit.github.core.client.GitHubClient.parseJson(GitHubClient.java:403)
    at org.eclipse.egit.github.core.client.GitHubClient.parseError(GitHubClient.java:514)
    at org.eclipse.egit.github.core.client.GitHubClient.createException(GitHubClient.java:547)
    at org.eclipse.egit.github.core.client.GitHubClient.get(GitHubClient.java:740)
    at org.eclipse.egit.github.core.client.PageIterator.next(PageIterator.java:173)
    at org.eclipse.egit.github.core.service.GitHubService.getAll(GitHubService.java:151)
    at org.eclipse.egit.github.core.service.PullRequestService.getPullRequests(PullRequestService.java:152)
    at com.carlosbecker.MainIntegrator.work(MainIntegrator.java:52)
    at com.carlosbecker.MainIntegrator.work(MainIntegrator.java:48)
    at com.carlosbecker.App.runOnce(App.java:31)
    at com.carlosbecker.App.run(App.java:25)
    at com.carlosbecker.App.main(App.java:38)
Caused by: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
    at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:187)
    at com.google.gson.Gson.fromJson(Gson.java:805)
    at com.google.gson.Gson.fromJson(Gson.java:770)
    at org.eclipse.egit.github.core.client.GitHubClient.parseJson(GitHubClient.java:422)
    ... 12 more
Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
    at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:387)
    at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:176)
    ... 15 more
```

Happens only when the something is configured wrong in the executions.json (e.g.: the repository doesn't exist)

--- a/src/main/java/com/carlosbecker/App.java
+++ b/src/main/java/com/carlosbecker/App.java
@@ -1,13 +1,12 @@
 package com.carlosbecker;
 
 import static com.google.inject.Guice.createInjector;
+import com.carlosbecker.github.IntegratorConfig;
 import java.io.File;
 import java.io.FileInputStream;
-import java.io.IOException;
 import javax.inject.Inject;
 import lombok.NonNull;
 import lombok.extern.log4j.Log4j;
-import com.carlosbecker.github.IntegratorConfig;
 
 @Log4j
 public class App {
@@ -26,7 +25,7 @@ public class App {
         } while (config.loop());
     }
 
-    private void runOnce() throws IOException, InterruptedException {
+    private void runOnce() throws InterruptedException {
         log.info("Running...");
         integrator.work();
         Thread.sleep(config.period() * 1000);

--- a/src/main/java/com/carlosbecker/MainIntegrator.java
+++ b/src/main/java/com/carlosbecker/MainIntegrator.java
@@ -42,10 +42,17 @@ public class MainIntegrator {
         this.executor = executor;
     }
 
-    public void work() throws IOException {
+    public void work() {
         final Iterator<ScriptedRepository> iterator = repositories.iterator();
-        while (iterator.hasNext())
-            work(iterator.next());
+        while (iterator.hasNext()) {
+            final ScriptedRepository repository = iterator.next();
+            try {
+                work(repository);
+            } catch (final IOException e) {
+                log.info(String.format("Failed to process repository '%s' due to a network or github error.",
+                        repository.getId()), e);
+            }
+        }
     }
 
     private void work(ScriptedRepository repository) throws IOException {

--- a/src/test/java/com/carlosbecker/AppTest.java
+++ b/src/test/java/com/carlosbecker/AppTest.java
@@ -1,13 +1,12 @@
 package com.carlosbecker;
 
 import static org.mockito.MockitoAnnotations.initMocks;
-import java.util.concurrent.atomic.AtomicBoolean;
-import org.eclipse.egit.github.core.client.RequestException;
+import com.carlosbecker.github.IntegratorConfig;
 import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mock;
 import org.mockito.Mockito;
-import com.carlosbecker.github.IntegratorConfig;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 public class AppTest {
     private App app;
@@ -42,10 +41,5 @@ public class AppTest {
     @Test(expected = NullPointerException.class)
     public void testMainMethodWithNullArgument() throws Exception {
         App.main(new String[] { null });
-    }
-
-    @Test(expected = RequestException.class)
-    public void testMainMethod() throws Exception {
-        App.main(new String[] { "./src/test/resources/main.test.properties" });
     }
 }

--- a/src/test/java/com/carlosbecker/MainIntegratorTest.java
+++ b/src/test/java/com/carlosbecker/MainIntegratorTest.java
@@ -54,7 +54,7 @@ public class MainIntegratorTest {
     @Test
     public void testPRWithNoComments() throws Exception {
         mockPullRequest();
-        mockRepositories();
+        mockRepository();
         when(issueService.getComments(any(IRepositoryIdProvider.class), eq(1))).thenReturn(newArrayList());
         integrator.work();
         verifyZeroInteractions(executor);
@@ -63,7 +63,7 @@ public class MainIntegratorTest {
     @Test
     public void testPRWithEmptyComment() throws Exception {
         mockPullRequest();
-        mockRepositories();
+        mockRepository();
         mockComments("");
         integrator.work();
         verifyZeroInteractions(executor);
@@ -72,7 +72,7 @@ public class MainIntegratorTest {
     @Test
     public void testWithNonMatchingComment() throws Exception {
         mockPullRequest();
-        mockRepositories();
+        mockRepository();
         mockComments("This will not match. Don't do it.");
         integrator.work();
         verifyZeroInteractions(executor);
@@ -81,7 +81,7 @@ public class MainIntegratorTest {
     @Test
     public void testMatchingComments() throws Exception {
         mockPullRequest();
-        mockRepositories();
+        mockRepository();
         mockComments("do it", "do IT");
         integrator.work();
         verify(executor).execute("echo", asList("user", "repo", "feature/my-branch", "1"));
@@ -90,7 +90,7 @@ public class MainIntegratorTest {
     @Test
     public void testWhenOneWasAlreadyExecuted() throws Exception {
         mockPullRequest();
-        mockRepositories();
+        mockRepository();
         mockComments("do it", "Ok, working on 'do it'...");
         integrator.work();
         verifyZeroInteractions(executor);
@@ -99,7 +99,7 @@ public class MainIntegratorTest {
     @Test
     public void testWhenOneWasAlreadyExecutedWithDifferentParams() throws Exception {
         mockPullRequest();
-        mockRepositories("do (it|that)");
+        mockRepository("do (it|that)");
         mockComments("do it", "Ok, working on 'do it'...", "Ok, working on 'do it'...", "do that");
         integrator.work();
         verify(executor).execute("echo", asList("user", "repo", "feature/my-branch", "1", "that"));
@@ -109,11 +109,20 @@ public class MainIntegratorTest {
     @Test(expected = RuntimeException.class)
     public void testCommentError() throws Exception {
         mockPullRequest();
-        mockRepositories();
+        mockRepository();
         mockComments("do it");
         when(issueService.createComment(eq("user"), eq("repo"), eq(1), eq("Ok, working on 'do it'..."))).thenThrow(
                 IOException.class);
         integrator.work();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testGithubParseError() throws Exception {
+        final ScriptedRepository repo = mockRepository();
+        when(prService.getPullRequests(eq(repo.getId()), eq("open"))).thenThrow(IOException.class);
+        integrator.work();
+        verifyZeroInteractions(issueService);
     }
 
     private void mockComments(String... messages) throws IOException {
@@ -126,13 +135,15 @@ public class MainIntegratorTest {
         when(issueService.getComments(any(IRepositoryIdProvider.class), eq(1))).thenReturn(comments);
     }
 
-    private void mockRepositories(String regex) {
+    private ScriptedRepository mockRepository(String regex) {
+        final ScriptedRepository repository = new ScriptedRepository("user", "repo", regex, "echo");
         when(repositories.iterator()).thenReturn(
-                asList(new ScriptedRepository("user", "repo", regex, "echo")).iterator());
+                asList(repository).iterator());
+        return repository;
     }
 
-    private void mockRepositories() {
-        mockRepositories("do it");
+    private ScriptedRepository mockRepository() {
+        return mockRepository("do it");
     }
 
     private void mockPullRequest() throws IOException {

